src/vivarium_census_prl_synth_pop/components/population.py

```diff
@@ -630,6 +631,214 @@ def assign_second_guardian_to_newborns(
 
         return new_births
 
+    def assign_college_simulants_guaridans(self, pop: pd.DataFrame) -> pd.DataFrame:
+        # Takes pop (simulation state table) and updates guardian_1 and guardian_2 atributes for college GQ simulants.
+
+        college_sims = pop.loc[
+            (pop["household_id"] == data_values.NONINSTITUTIONAL_GROUP_QUARTER_IDS["College"]) &
+            (pop["age"] < 24),
+            ["race_ethnicity"],
+        ]
+        # Get subsets of pop to create data structure with hhousehold_id, reference_person_id, partner_id, sex, race
+        #  for easier lookups
+        reference_persons = (
+            pop.loc[
+                (pop["relation_to_household_head"] == "Reference person") &
+                (pop["age"] >= 35) &
+                (pop["age"] < 66),
+                ["household_id", "sex", "race_ethnicity"],
+            ]
+            .reset_index()
+            .set_index("household_id")
+            .rename(columns={"index": "reference_person_id"})
+        )
+        partners_of_reference_persons = (
+            pop.loc[pop["relation_to_household_head"].isin(partners), ["household_id"]]
+            .reset_index()
+            .set_index("household_id")
+            .rename(columns={"index": "partner_id"})
+        )
```

Comment:
I've have updated this to choose the random partner here if there are multiple ones in a household.  I am curious why we want to handle this here instead of later?   When we did guardian initialization for children we handled it later but I am assuming that is because it was more complicated.